### PR TITLE
feat(storage): add transfer-acceleration to js v6

### PR DESCRIPTION
**Pull request by ashwinkumar6**
#### Description of changes:
Add the following changes to Js lib-v1 : Storage
- Add transfer acceleration page to storage
- Fix upload data page
- Remove `targetIdentityId` option from uploadData API 

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib-v1/storage/js/upload.mdx [hunk 1]
```diff
@@ -11,7 +11,6 @@ try {
 		data: file,
 		options: {
 			accessLevel: 'guest', // defaults to `guest` but can be 'private' | 'protected' | 'guest'
-			targetIdentityId: 'XXXXXXX', // id of another user, if `accessLevel: protected`
```
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/upload.mdx:14]: Good catch!

File: src/fragments/lib-v1/storage/js/transfer-acceleration.mdx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+}
+```
+
+Next, deploy this storage resource:
+
+```sh
+amplify push
+```
+
+
```
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/transfer-acceleration.mdx:39]:
```suggestion
```
extra lines?

File: src/pages/lib-v1/storage/upload/q/platform/[platform].mdx [hunk 3]
```diff
@@ -3,6 +3,10 @@ export const meta = {
   description: `Learn more about how to upload files using Amplify's storage category`
 };
 
+import js0 from '/src/fragments/lib-v1/storage/js/upload.mdx';
+
+<Fragments fragments={{ js: js0, "react-native": js0 }} />
```
Comment (AllanZhengYP) [inline src/pages/lib-v1/storage/upload/q/platform/[platform].mdx:8]: Ditto
```suggestion
<Fragments fragments={{ js: js0 }} />
```

File: src/pages/lib-v1/storage/transfer-acceleration/q/platform/[platform].mdx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+export const meta = {
+  title: `Use Transfer Acceleration`,
+  description: `How to enable and use S3's Transfer Acceleration on your S3 bucket`,
+};
+
+import js from "/src/fragments/lib-v1/storage/js/transfer-acceleration.mdx";
+
+<Fragments fragments={{js: js, 'react-native': js}} />
```
[the remark this answers is not in the document]
Comment (ashwinkumar6) [reply src/pages/lib-v1/storage/transfer-acceleration/q/platform/[platform].mdx:8]: Good catch!

File: src/pages/lib-v1/storage/upload/q/platform/[platform].mdx (hunk 3 above)
Comment (ashwinkumar6) [reply src/pages/lib-v1/storage/upload/q/platform/[platform].mdx]: Good catch!

File: src/fragments/lib-v1/storage/js/transfer-acceleration.mdx (hunk 2 above)
Comment (ashwinkumar6) [reply src/fragments/lib-v1/storage/js/transfer-acceleration.mdx:39]: thank you

File: src/fragments/lib-v1/storage/js/upload.mdx (hunk 1 above)
Comment (ashwinkumar6) [reply src/fragments/lib-v1/storage/js/upload.mdx:14]: thank you

File: src/directory/directory.mjs [hunk 5]
```diff
@@ -1467,6 +1467,11 @@ export const directory = {
             route: '/lib-v1/storage/configureaccess',
             filters: ['android', 'ios', 'flutter', 'js']
           },
+          {
+            title: 'Use Transfer Acceleration',
+            route: '/lib-v1/storage/transfer-acceleration',
+            filters: ['android', 'js', 'react-native', 'flutter']
```
Comment (jimblanc) [inline src/directory/directory.mjs:1473]: If this is an entirely new page for the `lib-v1` route, shouldn't we only add it to JS?

File: src/pages/lib-v1/storage/transfer-acceleration/q/platform/[platform].mdx [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+<Fragments fragments={{js: js}} />
+
+import androidDoc from '/src/fragments/lib/storage/native_common/transfer-acceleration/common.mdx';
+
+<Fragments fragments={{ android: androidDoc }} />
+
+import flutterDoc from '/src/fragments/lib/storage/native_common/transfer-acceleration/common.mdx';
+
+<Fragments fragments={{ flutter: flutterDoc }} />
```
Comment (jimblanc) [inline src/pages/lib-v1/storage/transfer-acceleration/q/platform/[platform].mdx:16]: Ditto, I think we should avoid adding new content to other platform's `lib-v1` routes

File: src/directory/directory.mjs (hunk 5 above)
Comment (ashwinkumar6) [reply src/directory/directory.mjs:1473]: We have this page in [V5 as well](https://github.com/aws-amplify/docs/pull/5906/files/66f04742df2707d711a641ff2d8718d006ec173b#diff-782c64a4a6ba1f0bff31bf639532b0169bd826b3e4208dc69ed6af06a98b08aeL797). I've just updated the route

File: src/pages/lib-v1/storage/transfer-acceleration/q/platform/[platform].mdx (hunk 6 above)
Comment (ashwinkumar6) [reply src/pages/lib-v1/storage/transfer-acceleration/q/platform/[platform].mdx:16]: Took an [existing reference for this as well from V5](https://github.com/aws-amplify/docs/blob/main/src/pages/lib/storage/transfer-acceleration/q/platform/%5Bplatform%5D.mdx)

File: src/directory/directory.mjs (hunk 5 above)
Comment (ashwinkumar6) [reply src/directory/directory.mjs]: ahh makes sense, updated

File: src/pages/lib-v1/storage/transfer-acceleration/q/platform/[platform].mdx (hunk 6 above)
Comment (ashwinkumar6) [reply src/pages/lib-v1/storage/transfer-acceleration/q/platform/[platform].mdx]: ahh makes sense, updated

**Review by AllanZhengYP — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]
Looks good!

**Pull request merged**